eng/pipelines/evaluate-changed-paths.sh

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env bash
+: '
+Scenarios:
+  1. exclude paths are specified
+      Will include all paths except the ones in the exclude list.
+  2. include paths are specified
+      Will only include paths specified in the list.
+  3. exclude + include:
+      1st we evaluate changes for all paths except ones in excluded list. If we can not find
+      any applicable changes like that, then we evaluate changes for incldued paths
+      if any of these two finds changes, then a variable will be set to true.
+  In order to consume this variable you need to reference it via: $[ dependencies.checkout.outputs["SetPathVars_<subset>.containschange"] ]
+
+  Example:
+  -difftarget ''HEAD^1'' -subset coreclr -includepaths ''src/libraries/System.Private.CoreLib/*'' -excludepaths ''src/libraries/*+src/installer/*''
+
+  This example will include ALL path changes except the ones under src/libraries/*!System.Private.CoreLib/*
+'
+
+# Disable globbing in this bash script since we iterate over path patterns
+set -f
+
+usage()
+{
+  echo "Script that evaluates changed paths and emits an azure devops variable if the changes contained in the current HEAD against the difftarget meet the includepahts/excludepaths filters:"
+  echo "  --difftarget <value>       SHA or branch to diff against. (i.e: HEAD^1, origin/master, 0f4hd36, etc.)"
+  echo "  --excludepaths <value>     Escaped list of paths to exclude from diff separated by '+'. (i.e: 'src/libraries/*+'src/installer/*')"
+  echo "  --includepaths <value>     Escaped list of paths to include on diff separated by '+'. (i.e: 'src/libraries/System.Private.CoreLib/*')"
+  echo "  --subset                   Subset name for which we're evaluating in order to include it in logs"
+  echo "  --azurevariable            Name of azure devops variable to create if change meets filter criteria (Default: containschange)"
```

Comment:
The help here specifies `--` prefix on the arguments yet all of the YML is using `-`

Reply:
Yeah. This was following the argument parsing from arcade which accepts both: `-` and `--` for consistency in between powershell and bash and to support bash format for args which is `--`. 

https://github.com/dotnet/runtime/pull/748/files#diff-47279b0209e4daf97bf498d42a8bc51fR33

I can update the caller to be consistent.

Reply:
Updated the .yml file to use `--`